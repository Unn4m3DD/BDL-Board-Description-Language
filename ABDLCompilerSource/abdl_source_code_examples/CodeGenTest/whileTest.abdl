main:
    let i = 2;
    let x = 1;
    while x < 10 do
        let i = 3;
        x = i + x;
    done;
    i = 10;
end main;