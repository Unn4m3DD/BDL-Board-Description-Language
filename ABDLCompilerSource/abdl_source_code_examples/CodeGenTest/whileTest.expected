import ABDLVar from "./AbdlVar.js"
var context;
var table;
var render;
const sleep = (milliseconds) => {
  return new Promise(resolve => setTimeout(resolve, milliseconds))
}
const range = (a, b) => {
  const result = [];
  for (let i = a.getValue(); i <= b.getValue(); i++) result.push(new ABDLVar(i));
  return result;
}
const getName = (a) => {
  try {
    return new ABDLVar(context.board[a.getValue()[0]][a.getValue()[1]].piece.name)
  } catch (e) {
    return new ABDLVar("");
  }
}
const getOwner = (a) => {
  try {
    return new ABDLVar(context.board[a.getValue()[0]][a.getValue()[1]].piece.owner)
  } catch (e) {
    return new ABDLVar(-1);
  }
}
const move = async (origin, destination) => {
  try {
    table[origin.getValue()[0]][origin.getValue()[1]].onclick();
    render(context, table)
    await sleep(1000);
    table[destination.getValue()[0]][destination.getValue()[1]].onclick();
    render(context, table)
    return new ABDLVar(1);
  } catch (e) {
    return new ABDLVar(0);
  }
}
const can_move = async (origin, destination) => {
  let result
  try{
    if (!table[origin.getValue()[0]][origin.getValue()[1]].onclick) return new ABDLVar(0);
      table[origin.getValue()[0]][origin.getValue()[1]].onclick();
  } catch(e) { return new ABDLVar(0); }
  render(context, table)
  try {
    if (
      table[destination.getValue()[0]][destination.getValue()[1]].style.backgroundColor ==
      "rgb(100, 200, 100)"
    )//rgb(100, 200, 100)
      result = new ABDLVar(1);
    else
      result = new ABDLVar(0);
  } catch (e) { result = new ABDLVar(0); }
  context.focused.x = -1;
  context.focused.y = -1;
  render(context, table);
  return result
}
export default {
  main: async (arg_context, arg_table, arg_render) => {
    context = arg_context;
    table = arg_table;
    render = arg_render;
    let v0 = new ABDLVar(2);
    let v1 = v0;
    let v2 = new ABDLVar(1);
    let v3 = v2;
    let v5 = new ABDLVar(10);
    let v4 = v3.less_than(v5);
    while (v4.getValue() != 0) {
      let v6 = new ABDLVar(3);
      let v7 = v6;
      let v8 = v7.add(v3);
      v3 = v8;
      v5 = new ABDLVar(10);
      v4 = v3.less_than(v5);

    }
    let v9 = new ABDLVar(10);
    v1 = v9;
  }
}
